syntax = "proto3";

service Showtime {
    rpc GetListShowtimes(EmptyShowTime) returns (stream ShowtimeData) {}
    rpc GetShowtimeByDate(ShowtimeDate) returns (ShowtimeData) {}
    rpc GetShowtimeByMovie(ShowtimeMovie) returns (ShowtimeDates) {}
}

message EmptyShowTime {}

message ShowtimeDate {
    string date = 1;
}

message ShowtimeMovie {
    string movie = 1;
}

message ShowtimeData {
    string date = 1;
    repeated Movie_list movies = 2;
}

message Movie_list {
    string movie = 1;
}

message ShowtimeDates {
  repeated string dates = 1;
}
